pandas/tests/io/test_common.py

```diff
@@ -89,6 +107,68 @@ def test_iterator(self):
         tm.assert_frame_equal(first, expected.iloc[[0]])
         tm.assert_frame_equal(concat(it), expected.iloc[1:])
 
+    @pytest.mark.parametrize('reader, module, path', [
```

Comment:
> though not sure whether I like better in each module, or centralized.

I like both :) Per-module for things specific to that, and centralized for "interface"-type things like this. The importskips are kind of annoying though. Happy to split this if you want.

I guess one alternative would be some kind of abstract base class for this set of tests, that would require methods to "implement" tests for `test_fspath`, `test_compression`, etc. That seems like a bit much to me though.